components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/DCRMConstants.java

```diff
@@ -48,20 +48,35 @@
         CONFLICT_EXISTING_CLIENT_ID("Client id %s already exist in the system"),
         BAD_REQUEST_CLIENT_ID_VIOLATES_PATTERN("Provided client id is not adhering to the provided regex %s"),
         FORBIDDEN_UNAUTHORIZED_USER("User does not have access to the application %s"),
-        ERROR_CODE_UNEXPECTED("Unexpected error");
+        ERROR_CODE_UNEXPECTED("Unexpected error"),
+        TENANT_DOMAIN_MISMATCH("NOT_FOUND_60001", "Tenant domain in request does not match with the application " +
```

Comment:
Any reason why we are sending the error code in "NOT_FOUND_60001" format?

Reply:
it is the format used to generate HTTP error response to the client
https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/main/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/util/DCRMUtils.java#L98

Reply:
otherwise 500 internal server error will be returned by default for all the cases if we don't add an error code

Reply:
Can we create a git issue to improve this?
It looks like a bit of a hack to pass the http status code 😃

Reply:
created an issue for that https://github.com/wso2/product-is/issues/8615